### DISCUSSION TITLE
R version support for PACE-hrh

### DISCUSSION THREAD

**Discussion by MeWu-IDM** [InstituteforDiseaseModeling/PACE-HRH#283] [Ideas]
Discussed briefly with @celiot-IDM 
Currently the automated build and testing were using R 4.2 version
although we do not anticipate imcompatibility for older/newer versions but may be a good practice to  declare a dependency to a minimum R version in DESCRIPTION file.
also we should discuss if build/test need to be updated for future R versions (e.g. 4.3)

**Comment by celiot-IDM** [comment]
I did some investigation on this topic, and found this passage in Hadley's book on R package development (at https://r-pkgs.org/description.html#depends-and-linkingto)

============

The most legitimate current use of Depends is to state a minimum version for R itself, e.g. Depends: R (>= 4.0.0). Again, think carefully if you do this. This raises the same issues as setting a minimum version for a package you depend on, except the stakes are much higher when it comes to R itself. Users can’t simply consent to the necessary upgrade, so, if other packages depend on yours, your minimum version requirement for R can cause a cascade of package installation failures.
 
The [backports package](https://cran.r-project.org/package=backports) is useful if you want to use a function like [tools::R_user_dir()](https://rdrr.io/r/tools/userdir.html), which was introduced in 4.0.0 in 2020, while still supporting older R versions.
 
The tidyverse packages officially support the current R version, the devel version, and four previous versions.[3](https://r-pkgs.org/description.html#fn3) We proactively test this support in the standard build matrix we use for continuous integration. Packages with a lower level of use may not need this level of rigor. The main takeaway is: if you state a minimum of R, you should have a reason and you should take reasonable measures to test your claim regularly.

============

Before we commit to setting an R version dependency, let's pause for a moment and think about how that might impact pacehrh users (not just ourselves).
